pgsql-flexible-server.tf

```diff
@@ -141,25 +157,49 @@ resource "null_resource" "set-user-permissions-additionaldbs" {
   for_each = var.enable_read_only_group_access ? { for index, db in var.pgsql_databases : db.name => db } : {}
 
   triggers = {
-    script_hash    = filesha256("${path.module}/set-postgres-permissions.bash")
-    name           = local.name
-    db_reader_user = local.db_reader_user
+    script_hash       = filesha256("${path.module}/set-postgres-permissions.bash")
+    name              = local.name
+    db_reader_user    = local.db_reader_user
+    add_sdp_read_user = var.add_sdp_read_user
+    sdp_reader_user   = local.sdp_read_user
   }
 
   provisioner "local-exec" {
     command = "${path.module}/set-postgres-permissions.bash"
 
     environment = {
-      DB_HOST_NAME   = azurerm_postgresql_flexible_server.pgsql_server.fqdn
-      DB_USER        = data.azuread_service_principal.mi_name[0].display_name
-      DB_READER_USER = local.db_reader_user
-      DB_NAME        = each.value.name
-      DB_ADMIN       = azurerm_postgresql_flexible_server.pgsql_server.administrator_login
-      DB_PASSWORD    = azurerm_postgresql_flexible_server.pgsql_server.administrator_password
+      DB_HOST_NAME    = azurerm_postgresql_flexible_server.pgsql_server.fqdn
+      DB_USER         = data.azuread_service_principal.mi_name[0].display_name
+      DB_READER_USER  = local.db_reader_user
+      DB_NAME         = each.value.name
+      DB_ADMIN        = azurerm_postgresql_flexible_server.pgsql_server.administrator_login
+      DB_PASSWORD     = azurerm_postgresql_flexible_server.pgsql_server.administrator_password
+      DB_ADD_SDP_USER = var.add_sdp_read_user
+      DB_SDP_USER     = local.sdp_read_user
+      DB_SDP_PASS     = random_password.sdp_read_user_password.result
     }
   }
   depends_on = [
     azurerm_postgresql_flexible_server_active_directory_administrator.pgsql_principal_admin,
     azurerm_postgresql_flexible_server_database.pg_databases
   ]
 }
+
+data "azurerm_key_vault" "sdp_vault" {
```

Comment:
This won't work unfortunately as terraform is scoped to a subscription and will only act in the active subscription, this would work in SDS but not CFT.

Of course you can pass a different provider to e.g. azurerm_mi, but getting teams to pass a new provider is a breaking change and one we wouldn't want to do for something niche like sdp from our generic module.

Your 2 options that I can think of are:
1. Create a dedicated module just for setting up the user and vault, that can have the specific providers setup from the beginning and would be a very turnkey solution.
2. Keep it in here but output the password from the module and get consumers to store it in the vault for you from their terraform

Both options I think would be quite straightforward with 1. probably being the cleanest for services, thoughts?

Reply:
Thanks @timja, will try option 1 with our own dedicated module.